### DISCUSSION TITLE
software interface pc what is suggest for the custom firmware

### DISCUSSION THREAD

**Discussion by joc2** [egzumer/uv-k5-firmware-custom#176] [Q&A]
The memory mapping of option( feature)  in the original quansheng is different then this great upgrade :)   

I explain by a quick example:
 You have 2 programable key, that have 2 setting ( quick and long push) originaly
in the custom version, you get 3 programable key , the 2 same as the original and it add menu key (long press). ( that is great, i love this option)

but if read the radio via chirp (last version), i only get 2 programable key, not the menu key.!!!

The question is also , if you write data from chirp to the radio, what will be set for the menu option missing in chirp ???. is the feature menu will be overwritten by something else or leave blank.

is their a way to adapte chirp or other pc program (portable radio cps) the get the adaptable memory map to follow new option(feature) 

their other option with the same isue, but this was the esieast one to describe.

let me know, what is involve and if i can help 
 Thank's a a nice day  :)  
![chirp reading](https://github.com/egzumer/uv-k5-firmware-custom/assets/56229329/773a588a-c3ac-4b4c-9363-cad2aa0a336b)

**Comment by joc2** [comment]
i found the driver in chirp   
https://chirp.danplanet.com/projects/chirp/repository/github/revisions/master/entry/chirp/drivers/uvk5.py

can you provide me the mapping for the software you did ex version 019...   

i will look to make then match if possible  
let me know

**Comment by joc2** [comment]
i got the memory map in the setting file to adapt it at chirp. i have contact chirp either and they can do it  but i need your agreement to do it, so it can be implement in chirp a version of your new mapping.   ( key menu, backlight level, etc. ) let me know

**Comment by ErikS-web** [comment]
See #183
